pkg/importer/imageio-datasource.go

```diff
@@ -18,626 +18,110 @@ package importer
-		ctx:           ctx,
-		cancel:        cancel,
-		imageioReader: imageioReader,
-		contentLength: contentLength,
-		imageTransfer: it,
-		connection:    conn,
-	}
-	// We know this is a counting reader, so no need to check.
-	countingReader := imageioReader.(*util.CountingReader)
-	go imageioSource.pollProgress(countingReader, 10*time.Minute, time.Second)
-
-	terminationChannel := newTerminationChannel()
-	go func() {
-		<-terminationChannel
-		klog.Infof("Caught termination signal, closing ImageIO.")
-		err := cancelTransfer(conn, it)
-		if err != nil {
-			klog.Errorf("Error cancelling transfer: %v", err)
-		}
-		err = imageioSource.Close()
-		if err != nil {
-			klog.Errorf("Error closing source: %v", err)
-		}
-	}()
+var mockOVirtClient = ovirtclient.NewMock()
 
-	return imageioSource, nil
+var mockOvirtClientFactory ovirtClientFactory = func(
+	endpoint string,
+	accessKey string,
+	secKey string,
+	certDir string,
+) (ovirtclient.Client, error) {
+	return mockOVirtClient, nil
 }
 
-// Info is called to get initial information about the data.
-func (is *ImageioDataSource) Info() (ProcessingPhase, error) {
-	var err error
-	is.readers, err = NewFormatReaders(is.imageioReader, is.contentLength)
-	if err != nil {
-		klog.Errorf("Error creating readers: %v", err)
-		return ProcessingPhaseError, err
-	}
+var newOVirtClient = defaultOVirtClientFactory
 
-	if !is.readers.Convert {
-		return ProcessingPhaseTransferDataFile, nil
-	}
-	return ProcessingPhaseTransferScratch, nil
-}
-
-// Transfer is called to transfer the data from the source to a scratch location.
-func (is *ImageioDataSource) Transfer(path string) (ProcessingPhase, error) {
-	// we know that there won't be archives
-	size, _ := util.GetAvailableSpace(path)
-	if size <= int64(0) {
-		//Path provided is invalid.
-		return ProcessingPhaseError, ErrInvalidPath
-	}
-	file := filepath.Join(path, tempFile)
-	err := util.StreamDataToFile(is.readers.TopReader(), file)
+// NewImageioDataSource creates a new instance of the ovirt-imageio data provider.
+func NewImageioDataSource(endpoint string, accessKey string, secKey string, certDir string, diskID string) (DataSourceInterface, error) {
+	cli, err := newOVirtClient(endpoint, accessKey, secKey, certDir)
 	if err != nil {
-		return ProcessingPhaseError, err
+		return nil, fmt.Errorf("failed to create oVirt client (%w)", err)
 	}
-	// If we successfully wrote to the file, then the parse will succeed.
-	is.url, _ = url.Parse(file)
-	return ProcessingPhaseConvert, nil
-}
 
-// TransferFile is called to transfer the data from the source to the passed in file.
-func (is *ImageioDataSource) TransferFile(fileName string) (ProcessingPhase, error) {
-	is.readers.StartProgressUpdate()
-	err := util.StreamDataToFile(is.readers.TopReader(), fileName)
+	ctx, cancel := context.WithCancel(context.Background())
+	download, err := cli.StartImageDownload(diskID, ovirtclient.ImageFormatRaw, ovirtclient.ContextStrategy(ctx))
```

Comment:
Any reason we are starting a download during the setup of the data processsor? We shouldn't start a download until we hit the Transfer function.